libevm/VMCalls.cpp

```diff
@@ -248,27 +250,26 @@ bool VM::caseCallSetup(CallParameters *callParams, bytesRef& o_output)
 	ON_OP();
 	updateIOGas();
 
-	if (m_OP != Instruction::DELEGATECALL && m_SP[2] > 0)
+	if (haveValueArg && m_SP[2] > 0)
 		callParams->gas += m_schedule->callStipend;
 
 	callParams->codeAddress = destinationAddr;
 
-	unsigned inOutOffset = 0;
-	if (m_OP == Instruction::DELEGATECALL || m_OP == Instruction::STATICCALL)
+	if (haveValueArg)
 	{
-		callParams->apparentValue = m_ext->value;
-		callParams->valueTransfer = 0;
+		callParams->valueTransfer = m_SP[2];
+		callParams->apparentValue = m_SP[2];
 	}
 	else
 	{
-		callParams->apparentValue = callParams->valueTransfer = m_SP[2];
-		inOutOffset = 1;
+		callParams->apparentValue = m_ext->value;
```

Comment:
I'm not sure what you mean by "this test". You referenced 2 tests before. If you mean https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/stStaticCall/static_callToStaticOpCodeCheckFiller.json#L43, this tests does other weird things like: checking for ORIGIN, multiple levels of different calls, etc. I'm not sure any of them check if the value from transaction is forwarded in STATICCALL.

Also, I don't know what you mean by "just add more values to the value list".